packages/react-components/react-table/src/components/DataGrid/useDataGrid.ts

```diff
@@ -17,7 +17,15 @@ export const useDataGrid_unstable = (props: DataGridProps, ref: React.Ref<HTMLEl
   const { items, columns, focusMode = 'none' } = props;
   const navigable = focusMode !== 'none';
   const keyboardNavAttr = useArrowNavigationGroup({ axis: 'grid' });
-  const tableState = useTable({ items, columns }, []);
+
+  const tableState = useTable({ items, columns }, [
+    useSort({
+      defaultSortState: props.defaultSortState,
+      sortState: props.sortState,
+      onSortChange: props.onSortChange as (e: React.SyntheticEvent, sortState: SortState) => void,
```

Comment:
not if removing this cast is possible, any suggestions welcome

Reply:
convert properties to methods and everything works fine 😁. I know it sounds like black magic... and it is.

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/5483269/202269843-2eef7b1f-bc45-402b-b1e3-23a8b4fa7b5c.png">

Reply:
this is crazy.. why does it work ???